### PR TITLE
Prompt users when navigating away with unsaved changes

### PR DESCRIPTION
**Pull request by spadgett**
Fixes https://github.com/openshift/origin/issues/6994

### CODE REVIEW

**Comment by spadgett** [comment]
Rebased

**Comment by spadgett** [comment]
Rebased. @jwforres Should we open a Trello card for this?

**Comment by spadgett** [comment]
Rebased

**Comment by spadgett** [comment]
Rebased

File: app/views/create-route.html [hunk 1]
```diff
@@ -40,7 +41,9 @@
                               ng-click="createRoute()"
                               ng-disabled="createRouteForm.$invalid || disableInputs || !createRoute"
                               value="">Create</button>
-                          <a class="btn btn-default btn-lg" href="#" back>Cancel</a>
+                          <a class="btn btn-default btn-lg"
+                              ng-click="doneEditing = true"
```
Comment (jwforres) [inline app/views/create-route.html]: we should use dot notation instead of setting a top level scope var since some of these are underneath ng-ifs

**Comment by jwforres** [comment]
and just adding a reminder for what we talked about around checking process template in a dialog to make sure closing the dialog with unsaved changes and then navigating away later doesn't cause the alert to trigger

File: app/views/create-route.html (hunk 1 above)
Comment (spadgett) [inline app/views/create-route.html]: @jwforres Confirmed it does incorrectly prompt on this form. Thanks for catching this.

**Comment by spadgett** [comment]
@jwforres Updated. I renamed `$scope.doneEditing` to `$scope.confirm.doneEditing`

**Comment by jwforres** [comment]
LGTM, needs rebasing. should try and get this merged today so QE has plenty of time to catch regressions

**Comment by spadgett** [comment]
@jwforres rebased

[test]

**Comment by jwforres** [comment]
[merge]

**Comment by spadgett** [comment]
Rebased

[merge]

**Pull request merged**